### PR TITLE
feat: support dynamic result returning generic type

### PR DESCRIPTION
**Pull request by YongwuHe**

### CODE REVIEW

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/model/DynamicClassEntity.java [hunk 1]
```diff
@@ -32,9 +37,22 @@ public DynamicClassEntity(String clazzName, String operation, String parameterTy
 
         // fix dynamic response type is generic type
         this.keyFormula = keyFormula;
+        this.isGenericReturnType = whetherGenericReturnType();
         this.returnType = null;
     }
 
+    private boolean whetherGenericReturnType() {
```
Comment (mr3) [inline arex-instrumentation-api/src/main/java/io/arex/inst/runtime/model/DynamicClassEntity.java:44]: renaming: whetherGenericReturnType to isGenericReturnType

File: arex-instrumentation/dynamic/arex-dynamic/src/main/java/io/arex/inst/dynamic/DynamicClassExtractor.java [hunk 2]
```diff
@@ -60,10 +63,31 @@ public void setFutureResponse(Future<?> result) {
...
+           return null;
+        }
+
+        if (resultClazz.contains(TypeUtil.HORIZONTAL_LINE_STR)) {
+            return resultClazz;
+        }
+
+        for (DynamicClassEntity entity : Config.get().getGenericReturnTypeEntities()) {
+           if (Objects.equals(clazzName, entity.getClazzName()) && Objects.equals(operation, entity.getOperation())) {
+               return resultClazz + TypeUtil.HORIZONTAL_LINE + entity.getKeyFormula();
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic/src/main/java/io/arex/inst/dynamic/DynamicClassExtractor.java:85]: Add static field: genericReturnTypeCacheMap, read from cache first

File: arex-instrumentation-api/src/main/java/io/arex/inst/runtime/config/Config.java [hunk 3]
```diff
@@ -34,6 +37,7 @@ public static Config get() {
         this.enableDebug = enableDebug;
         this.serviceName = serviceName;
         this.entities = entities;
+        this.genericReturnTypeEntities = entities == null ? Collections.emptyList() : entities.stream().filter(DynamicClassEntity::isGenericReturnType).collect(Collectors.toList());
```
Comment (lucas-myx) [inline arex-instrumentation-api/src/main/java/io/arex/inst/runtime/config/Config.java:40]: This filtering logic can be placed in `getGenericReturnTypeEntities()`, not need to define the `genericReturnTypeEntities` variable and reduces the complexity of the constructor.

File: arex-instrumentation/dynamic/arex-dynamic/src/main/java/io/arex/inst/dynamic/DynamicClassInstrumentation.java [hunk 4]
```diff
@@ -44,11 +44,11 @@ public DynamicClassInstrumentation(List<DynamicClassEntity> dynamicClassList) {
         this.dynamicClassList = dynamicClassList;
         for (DynamicClassEntity entity : dynamicClassList) {
             if (StringUtil.isNotEmpty(entity.getKeyFormula()) && StringUtil.isNotEmpty(entity.getOperation())) {
-                if (entity.getKeyFormula().contains("java.lang.System.currentTimeMillis")) {
+                if (entity.getKeyFormula().contains(DynamicClassEntity.CURRENT_TIME_MILLIS_SIGNATURE)) {
```
Comment (lucas-myx) [inline arex-instrumentation/dynamic/arex-dynamic/src/main/java/io/arex/inst/dynamic/DynamicClassInstrumentation.java:47]: The definition of constants should be maintained in `ArexConstants.java`

**Review by lucas-myx — APPROVED** [review]

**Review by mr3 — APPROVED** [review]

**Pull request merged**